logger/logger.go

```diff
@@ -239,6 +309,10 @@ func (l *ZapLogger) WithoutSampler() Logger {
 	return &dup
 }
 
+func (l *ZapLogger) isEnabled(level zapcore.Level) bool {
+	return l.sharedConfig.isEnabled(l.component, level)
```

Comment:
I am possibly not understanding this correctly. Is it possible to update loggers with component level if the logger has a component. This one is doing a sync map load on every log line. Feels like that could be expensive. Even if not expensive, feels like this logger struct can store the level based on component and just do a simple level check?

Reply:
we could probably load the log levels into something like `map[component]zap.AtomicLogLevel` then load the `zap.AtomicLogLevel` once when the logger is created and use `(zap.AtomicLogLevel).SetLevel` when the config updates. then we wouldn't need the config callback thing or custom level filtering. 

any log levels from the map unspecified by the updated config would revert to the default level. eliminating the callback list also removes the risk of a memory leak when creating these loggers outside of component init.